ibodata/lateral_profile.py

```diff
@@ -23,48 +23,79 @@ def penumbra_left(self):
     def field_ratio(self, level):
         """
         In case of good data returns positive number
+        Level has to be >= 0.0 and <= 1 or exception is raised
         In case of corrupted data returns nan
         """
+        if level < 0.0 or level > 1.0:
+            raise ValueError("Expected level to be between 0.0 and 1.0")
         return self.width(level) / self.width(0.5)
 
     def symmetry(self, level):
         """
         In case of good data returns positive number
+        Level has to be >= 0 and <= 1 or exception is raised
         In case of corrupted data returns nan
         """
+        if level < 0 or level > 1.0:
+            raise ValueError("Expected level to be between 0 and 1")
         a = math.fabs(self.x_at_y(level, False))
         b = math.fabs(self.x_at_y(level, True))
         return (math.fabs(a - b) / (a + b)) * 200.0
 
     def flatness_50(self):
+        """
+        In case of corrupted data returns nan
+        Returns value between points if max/min value occurs on border
+        """
         d = (self.penumbra_left() + self.penumbra_right()) / 2
-        if d == np.nan:
+        if d != d:
```

Comment:
Add a comment why such interesting comparison is being made. Is it a test for corrupted number ?
If yes, the please explain it.

Reply:
standard for representing floating-point numbers requires that NaN != NaN is always true. d != d is the same comparison as np.isnan(d)